lib/SILOptimizer/Transforms/CopyForwarding.cpp

```diff
@@ -1873,7 +1877,7 @@ bool TempRValueOptPass::tryOptimizeCopyIntoTemp(CopyAddrInst *copyInst) {
       continue;
 
     // Destroys and deallocations are allowed to be in a different block.
-    if (isa<DestroyAddrInst>(user) || isa<DeallocStackInst>(user))
+    if (isa<DestroyAddrInst>(user) || isa<DeallocStackInst>(user) || isa<MarkDependenceInst>(user))
```

Comment:
Don't we have to check the uses of the mark_dependence instruction?

Reply:
You are right. Looks like I do need to look at its uses, which will be an apply. And if the apply has an inout_aliasable we can't eliminate this copy_addr. If it had a in_guaranteed, there won't be a matching destroy_addr, and there's nothing to optimize. Looks like adding this pattern may not be useful at all!